com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1422,67 +1406,76 @@ internal void HandleApproval(ulong ownerClientId, bool createPlayerObject, uint?
-
-                                nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[ownerClientId].PlayerObject.transform.rotation.eulerAngles.x);
-                                nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[ownerClientId].PlayerObject.transform.rotation.eulerAngles.y);
-                                nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[ownerClientId].PlayerObject.transform.rotation.eulerAngles.z);
-                            }
-                            else
-                            {
-                                nonNullContext.NetworkWriter.WriteBool(false);
-                            }
+                        if (ConnectedClients[clientId].PlayerObject.IncludeTransformWhenSpawning == null || ConnectedClients[clientId].PlayerObject.IncludeTransformWhenSpawning(clientId))
+                        {
+                            nonNullContext.NetworkWriter.WriteBool(true);
+                            nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[clientId].PlayerObject.transform.position.x);
+                            nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[clientId].PlayerObject.transform.position.y);
+                            nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[clientId].PlayerObject.transform.position.z);
+
+                            nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[clientId].PlayerObject.transform.rotation.eulerAngles.x);
+                            nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[clientId].PlayerObject.transform.rotation.eulerAngles.y);
+                            nonNullContext.NetworkWriter.WriteSinglePacked(ConnectedClients[clientId].PlayerObject.transform.rotation.eulerAngles.z);
+                        }
+                        else
+                        {
+                            nonNullContext.NetworkWriter.WriteBool(false);
+                        }
 
-                            nonNullContext.NetworkWriter.WriteBool(false); //No payload data
+                        nonNullContext.NetworkWriter.WriteBool(false); //No payload data
 
-                            if (NetworkConfig.EnableNetworkVariable)
-                            {
-                                ConnectedClients[ownerClientId].PlayerObject.WriteNetworkVariableData(nonNullContext.NetworkWriter.GetStream(), clientPair.Key);
-                            }
+                        if (NetworkConfig.EnableNetworkVariable)
+                        {
+                            ConnectedClients[clientId].PlayerObject.WriteNetworkVariableData(nonNullContext.NetworkWriter.GetStream(), clientPair.Key);
                         }
                     }
                 }
             }
-            else
```

Comment:
Just for my understanding if we are not approved we no longer disconnect the connection is that expected or we now doing this elsewhere?

Reply:
no, it actually stays there but shifted around a little bit, see line 1414